java/src/main/java/org/apache/arrow/gandiva/evaluator/SelectionVector.java

```diff
@@ -49,6 +49,13 @@ public final int getRecordCount() {
     return this.recordCount;
   }
 
+  /*
+   * Release the buffer.
+   */
+  public final void close() {
```

Comment:
this looks wrong - since this object didn't take a ref, it shouldn't release.

The caller holds the ref on the buffer during the lifetime of the SelectionVector.